en/docs/guides/login/oidc-auth-code-pkce-public-clients.md

```diff
@@ -0,0 +1,135 @@
+
+        - To configure more advanced configurations, see [OAuth/OpenID Connect Configurations]({{base_path}}/guides/login/oauth-app-config-advanced).
+
+----
+
+## Get authorization code
+
+To obtain the `authorization_code`, send an authorization request to the authorization endpoint using the following request format.
+
+!!! tip
+    You can use this [Online PKCE Generator Tool](https://tonyxu-io.github.io/pkce-generator/) to generate PKCE code challenges.
+
+    To generate the **challenge** and **verifier** through the tool:
+
+    1. Click **Generate Code Verifier**.
+    2. Click **Generate Code Challenge**.
+    
+    Note the two values. You will need them to get the authorization code and the tokens. The code challenge you get here is the base64 URL encoded value of the SHA256 hashed `code_verifier`, so the code challenge method will be `S256`.
+
+**Request format**
+    ```bash
+    https://<host>/oauth2/authorize?scope=openid&response_type=code
+    &redirect_uri=<redirect_uri>
+    &client_id=<OAuth Client Key>
+    &code_challenge=<PKCE_code_challenge>
+    &code_challenge_method=<PKCE_code_challenge_method>
+    ```
+
+**Sample Request**
+    ```bash
+    https://localhost:9443/oauth2/authorize?scope=openid&response_type=code
+    &redirect_uri=https://localhost/callback
+    &client_id=YYVdAL3lLcmrubZ2IkflCAuLwk0a
+    &code_challenge=5vEtIy2T-G65yXHc8g5zcJDQXICBzZMrtERq0zhx7hM
+    &code_challenge_method=S256
+    ```
+
+This authorization request takes the following parameters.
+
+| Request Parameter | Description    |
+|-------------------|----------------|
+| `redirect_uri`  | This is where the response is redirected to at the end of the process. This needs to be the same as one of the URLs given in the registered apps. |
+| `client_id` | The client ID generated when registering the application on IS.
+| `code_challenge`    | The client creates and records a secret cryptographical random string (`code_verifier`), which is then encoded using URL-safe base64 encoding to transform it into the `code_challenge`. The `code_challenge` is required for the authorization code flow with PKCE. You can use some [tools](https://tonyxu-io.github.io/pkce-generator/) to generate the `code_challenge` and `code_verifier`.
+| `code_challenge_method` | This is the method used for transforming the `code_verifier` into the `code_challenge`. Identity Server supports S256 and `plain`. This is required for the authorization code flow with PKCE. |
+
+## Get tokens
+
+After receiving the authorization code, the application has to exchange it to get the following tokens:
+
+- `access_token`
+- `id_token`
+- `refresh_token` (only if the `refresh_token` grant type is enabled for the service provider registered on IS)
+
+You can get the tokens by sending the following token request to the token endpoint.
+
+**Request Format**
+    ```curl
+    curl -i -X POST -u <OAuth Client Key>:<Client Secret> 
```

Comment:
Hi @divyaamunugama ,

In this flow we don't use the client secret. Can you please refer this[1] and change this curl accordingly.

[1] https://medium.com/identity-beyond-borders/pkce-in-action-with-wso2-identity-server-d08f9106fd81#:~:text=Making%20a%20token%20request%20without%20a%20client%20secret